rosbag2_tests/test/memory_management.hpp

```diff
@@ -12,29 +12,20 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#ifndef ROSBAG2__TEST_MEMORY_MANAGEMENT_HPP_
-#define ROSBAG2__TEST_MEMORY_MANAGEMENT_HPP_
+#ifndef MEMORY_MANAGEMENT_HPP_
+#define MEMORY_MANAGEMENT_HPP_
```

Comment:
even though this is a test, the if guard should still be namespaced by the package name... ?

Reply:
Yes, linters... cpplint forces the header guard to reflect the folder structure, not the namespace or package structure. Don't ask me why.

But I agree it's more consistent and in other packages we also have the structure `<package_name>/tests/<package_name>` which enforces that all header guards start with `<PACKAGE_NAME>__`. I'll fix this for this package and all other places in rosbag2